src/typing/flow_js.ml

```diff
@@ -7351,6 +7407,7 @@ and eval_destructor cx ~trace use_op reason t d tout = match t with
   | ReadOnlyType ->
     let open Object in
     ObjKitT (use_op, reason, Resolve (Next), ReadOnly, tout)
+  | RequiredType -> RequiredT (reason, tout)
```

Comment:
Instead of adding a new use type, can you follow the way ReadOnly is implemented in object_kit and do the same? It will significantly simplify your code by handling a ton of edge cases for you.

For example, your code will error on mixed/empty, which the object_kit code can handle more elegantly.

You also won't need the IncompatibleRequiredT because object kit handles errors with types that can't be converted into an object.

You also shouldn't need to distinguish between instances and objects, since object_kit will take care of that for you.

Reply:
But object_kit doesn't have polarity

Reply:
If I change `and prop = TypeTerm.t * bool (* own *)` to `and prop = TypeTerm.t * bool (* own *) * TypeTerm.polarity` in type.ml, would it be okay in object_kit?

Reply:
Also there is issue in object_kit with statics, maybe you know how to fix it?
https://github.com/facebook/flow/issues/7572